tests/cupy_tests/test_cusparse.py

```diff
@@ -135,15 +134,75 @@ def test_csrgeam(self):
 
     def test_csrgeam2(self):
         if not cupy.cusparse.check_availability('csrgeam2'):
-            pytest.skip('csrgeam is not available')
-
+            pytest.skip('csrgeam2 is not available')
         a = sparse.csr_matrix(self.a)
         b = sparse.csr_matrix(self.b)
         c = cupy.cusparse.csrgeam2(a, b, alpha=self.alpha, beta=self.beta)
         expect = self.alpha * self.a + self.beta * self.b
         testing.assert_array_almost_equal(c.toarray(), expect.toarray())
 
 
+class TestCsrgeamInvalidCases(unittest.TestCase):
+
+    dtype = numpy.float32
+    shape = (4, 3)
+
+    def setUp(self):
+        m, n = self.shape
+        self.a = scipy.sparse.random(m, n, density=0.3, dtype=self.dtype)
+        self.b = scipy.sparse.random(m, n, density=0.3, dtype=self.dtype)
+
+    def test_csrgeam_invalid_format(self):
+        if not cupy.cusparse.check_availability('csrgeam'):
+            pytest.skip('csrgeam is not available')
+        a = sparse.csc_matrix(self.a)
+        b = sparse.csr_matrix(self.b)
+        with self.assertRaises(TypeError):
+            cupy.cusparse.csrgeam(a, b)
+        with self.assertRaises(TypeError):
+            cupy.cusparse.csrgeam(b, a)
+
+    def test_csrgeam_invalid_shape(self):
+        if not cupy.cusparse.check_availability('csrgeam'):
+            pytest.skip('csrgeam is not available')
+        a = sparse.csc_matrix(self.a).T
+        b = sparse.csr_matrix(self.b)
+        with self.assertRaises(ValueError):
+            cupy.cusparse.csrgeam(a, b)
+
+    def test_csrgeam_availability(self):
+        if not cupy.cusparse.check_availability('csrgeam'):
+            a = sparse.csr_matrix(self.a)
+            b = sparse.csr_matrix(self.b)
+            with self.assertRaises(RuntimeError):
+                cupy.cusparse.csrgeam(a, b)
+
+    def test_csrgeam2_invalid_format(self):
+        if not cupy.cusparse.check_availability('csrgeam2'):
+            pytest.skip('csrgeam2 is not available')
+        a = sparse.csc_matrix(self.a)
+        b = sparse.csr_matrix(self.b)
+        with self.assertRaises(TypeError):
+            cupy.cusparse.csrgeam2(a, b)
+        with self.assertRaises(TypeError):
+            cupy.cusparse.csrgeam2(b, a)
+
+    def test_csrgeam2_invalid_shape(self):
+        if not cupy.cusparse.check_availability('csrgeam2'):
+            pytest.skip('csrgeam2 is not available')
+        a = sparse.csc_matrix(self.a).T
+        b = sparse.csr_matrix(self.b)
```

Comment:
nit: shouldn't the matrices be the same types but transposed?

Reply:
Yes, both matrices `a` and `b` must be CSR format in `csrgeam2`.
In CuPy, when a CSC format matrix with shape of (m, n) is transposed, it becomes a CSR format matrix with shape of (n, m).
In this testing, there is no problem in terms of format (since both are CSR matrices), but it raises a ValueError due to shape mismatch.

Reply:
Maybe, `a = sparse.csr_matrix(self.a.T)` would be better?

Reply:
Yes, both of them csr but with different type 😅.
If the type and shape is different, the error would depend on the order of the internal checks 
in cuSparse, which is not guaranteed.

Thanks!